### PR TITLE
[APB-4216][SJM] Fixing error when user is not an agent and chooses to…

### PR DESCRIPTION
**Pull request by stevenmallen**
… create a new ID

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscriptionfrontend/config/frontendAppConfig.scala [hunk 1]
```diff
@@ -84,16 +82,24 @@ class FrontendAppConfig @Inject()(val environment: Environment, val configuratio
...
   override def agentMappingFrontendStartUrl(continueId: String): String =
     s"${getServicesConfStringOrFail("agent-mapping-frontend.external-url")}${getServicesConfStringOrFail(
       "agent-mapping-frontend.start.path")}?continueId=$continueId"
+
+  val ssoRedirectUrl: String = "/government-gateway-registration-frontend?accountType=agent&origin=unknown"
+
   override val ggRegistrationFrontendExternalUrl: String =
     s"${getConfStringOrFail(s"$env.microservice.services.government-gateway-registration-frontend.externalUrl")}$ssoRedirectUrl"
 
+  private val rootContinuePath: String = "/agent-subscription/return-after-gg-creds-created"
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/config/frontendAppConfig.scala]: could rename this returnAfterGGCredCreatedsPath or something like that

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignedOutController.scala [hunk 2]
```diff
@@ -40,26 +41,39 @@ class SignedOutController @Inject()(
...
+    withAuthenticatedUser {
+      for {
+        agentSubContinueUrlOpt <- sessionStoreService.fetchContinueUrl
+      } yield {
+        redirectToCreateCreds(agentSubContinueUrlOpt, None)
+      }
+    }
+  }
+
+  private def redirectToCreateCreds(maybeContinueUrl: Option[ContinueUrl], continueId: Option[String]): Result = {
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignedOutController.scala]: to fit the pattern of the above methods should be named redirectTocreateCleanCreds

File: it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala [hunk 3]
```diff
@@ -64,7 +64,7 @@ class SignOutControllerISpec extends BaseISpec {
       implicit val request: FakeRequest[AnyContentAsEmpty.type] = authenticatedAs(subscribingAgentEnrolledForNonMTD)
       sessionStoreService.currentSession.continueUrl = Some(ourContinueUrl)
 
-      val result = await(controller.redirectToSos(request))
+      val result = await(controller.redirectTaskListUserToCreateCleanCreds(request))
 
       val sosContinueValueUnencoded =
```
Comment (mariannepearson0) [inline it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala]: Any tests for the redirectUserToCreateCleanCreds method?

File: app/uk/gov/hmrc/agentsubscriptionfrontend/config/frontendAppConfig.scala (hunk 1 above)
Comment (stevenmallen) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/config/frontendAppConfig.scala]: Consider it done :)

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignedOutController.scala (hunk 2 above)
Comment (stevenmallen) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignedOutController.scala]: Yep

File: it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala (hunk 3 above)
Comment (stevenmallen) [reply it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SignOutControllerISpec.scala]: Making some now...

**Pull request merged**